DCRack/dc_rack_server.py

```diff
@@ -220,79 +295,71 @@ def identSelf(self, timeout=1):
-                             (card.C.highPass, card.C.lowPass, card.C.polarity, card.C.offset),
-                             (card.D.highPass, card.D.lowPass, card.D.polarity, card.D.offset)))
+            def state(chan):
+                """Return a tuple of channel state, to be stored in the registry"""
+                return (chan.highPass, chan.lowPass, chan.polarity, chan.offset)
+            p.set(cardName, (state(card.A), state(card.B), state(card.C), state(card.D)))
             yield p.send()
         else:
             print 'card is not a preamp'
-    
+
     @inlineCallbacks
     def loadFromRegistry(self, reg):
         card = self.rackCards[self.activeCard]
         if isinstance(card, Preamp):
             yield reg.cd(['', 'Servers', 'DC Racks', 'Preamps'], True)
-            cardName = 'Preamp ' + str(self.activeCard)
+            cardName = 'Preamp {}'.format(self.activeCard)
             p = reg.packet()
-            p.get(cardName, key = cardName)
+            p.get(cardName, key=cardName)
             result = yield p.send()
             ans = result[cardName]
-            card.A.highPass = ans[0][0]
-            card.A.lowPass = ans[0][1]
-            card.A.polarity = ans[0][2] 
-            card.A.offset = ans[0][3]
-            card.B.highPass = ans[1][0]
-            card.B.lowPass = ans[1][1]
-            card.B.polarity = ans[1][2]
-            card.B.offset = ans[1][3]
-            card.C.highPass = ans[2][0]
-            card.C.lowPass = ans[2][1]
-            card.C.polarity = ans[2][2]
-            card.C.offset = ans[2][3]
-            card.D.highPass = ans[3][0]
-            card.D.lowPass = ans[3][1]
-            card.D.polarity = ans[3][2]
-            card.D.offset = ans[3][3]
+            def update(chan, state):
+                """Update channel state from a tuple stored in the registry"""
+                chan.highPass, chan.lowPass, chan.polarity, chan.offset = state[0]
+            for i, chan in enumerate([card.A, card.B, card.C, card.D]):
+                update(chan, ans[i])
         else:
             print 'card is not a preamp'
 
 
     @inlineCallbacks
     def triggerChannel(self, channel):
-        """Tells selected channel to pull data from registry and update DAC value"""
-        ChannelID = {'A':0, 'B':1, 'C':2, 'D':3}[channel]
-        #Bitwise OR with 11000000
-        yield self.write([192|ChannelID])
+        """Tell the given channel to pull data from register and update DAC value"""
+        ID = CHANNEL_IDS[channel]
+        yield self.write([OP_TRIGGER | ID])
 
 
     @inlineCallbacks
-    def pushRegistryValue(self, dac, slow, voltage):
-        """Pushes 18 bits of data into 18 bit shift register. First bit is fine(0) or coarse(1) DAC, 
-        last bit is fast(0) or slow(1) slew rate, and middle 16 bits are voltage value"""
-        #Conversion of voltage value into 16-bit number, plus bits for DAC selection and slew rate
+    def pushRegisterValue(self, dac, slow, voltage):
```

Comment:
I was confused at first because the code uses 'registry' to refer to the labrad registry in other places, but here we're talking about the data *register* on the DAC card. I've tried to fix up the naming.